src/main/java/com/fauna/codec/json/QueryTagsDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.fauna.codec.json;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryTagsDeserializer extends JsonDeserializer<Map<String, String>> {
+    @Override
+    public Map<String,String> deserialize(JsonParser jp, DeserializationContext deserializationContext) throws IOException, JacksonException {
+        // We can improve performance by building this from tokens instead.
+        switch (jp.currentToken()) {
+            case VALUE_NULL:
+                return null;
+            case VALUE_STRING:
+                var raw = jp.getValueAsString();
+                Map<String, String> ret = new HashMap<>();
+                if (raw.isEmpty()) {
+                    return ret;
+                }
+
+                String[] tagPairs = jp.getValueAsString().split(",");
```

Comment:
Two calls to `jp.getValueAsString` seems unneccessary?

I think there's a faster and more readable way to implement this method, happy to do it in a future PR.

Reply:
Yep, this is supposed to be `raw.split(",")`

As for readability, happy to take specific suggestions if you'd like to make any.

Reply:
I'll follow up with a pr to clean this up